server/update_branch.go

```diff
@@ -11,40 +10,48 @@ import (
 	"github.com/google/go-github/v32/github"
 )
 
-var (
-	ErrCommenterPermission    = errors.New("commenter does not have permissions")
-	ErrOrganizationPermission = errors.New("we don't have permissions")
-	ErrUpdatePullRequest      = errors.New("could not update pull request")
-)
-
 const (
-	MsgCommenterPermission    = "Looks like you don't have permissions to trigger this command.\n Only available for the PR submitter and org members"
-	MsgOrganizationPermission = "We don't have permissions to update this PR, please contact the submitter to apply the update."
-	MsgUpdatePullRequest      = "Error trying to update the PR.\nPlease do it manually."
+	msgCommenterPermission    = "Looks like you don't have permissions to trigger this command.\n Only available for the PR submitter and org members"
+	msgOrganizationPermission = "We don't have permissions to update this PR, please contact the submitter to apply the update."
+	msgUpdatePullRequest      = "Error trying to update the PR.\nPlease do it manually."
 )
 
+type updateError struct {
+	source string
+}
+
+func (e *updateError) Error() string {
+	switch e.source {
+	case msgCommenterPermission:
+		return "commenter does not have permissions"
+	case msgOrganizationPermission:
+		return "we don't have permissions"
+	case msgUpdatePullRequest:
+		return "could not update pull request"
+	default:
+		panic("unhandled error type")
+	}
+}
+
 func (s *Server) handleUpdateBranch(ctx context.Context, commenter string, pr *model.PullRequest) error {
-	var err error
-	var msg string
+	var e *updateError
```

Comment:
Nit: `e` as a single letter is too short a variable name for the entire method. 0/5 on the new name, but a longer name would be more idiomatic IMO.

Reply:
The first longer name came into my mind was `ue` 😝